### ISSUE TITLE
Enable auto labels for PRs

### ISSUE THREAD

**Issue by liangyuanpeng** [karmada-io/karmada#5183]
**What would you like to be added**:

 let PR discovery to have them automatically labeled based on directory they touch, this is useful for people to filter PRs they are interested in, especially now that Summer OSPP 2024 is underway.

just enable the prow plugin of  `owners-label` and ` verify-owners` for it is engouth.

I have done this kind of work for etcd before. see https://github.com/kubernetes/test-infra/pull/32589

Basically, the label might look like this:
- area/karmadactl
- area/controller-manager
- area/helm charts
- area/scheduler
...

/assign
/cc @RainbowMango 

**Why is this needed**:
